libs/velo-external-db-core/src/service/schema.ts

```diff
@@ -61,12 +63,135 @@ export default class SchemaService {
         }))
     }
 
-    
     async validateOperation(operationName: SchemaOperations) {
         const allowedSchemaOperations = this.storage.supportedOperations()
 
         if (!allowedSchemaOperations.includes(operationName)) 
             throw new errors.UnsupportedOperation(`Your database doesn't support ${operationName} operation`)
     }
 
+    async listCollections(collectionIds: string[]): Promise<ListCollectionsResponsePart> {
+        
+        // remove in the end of development
+        if (!this.storage.capabilities || !this.storage.columnCapabilitiesFor) {
+            throw new Error('Your storage does not support the new collection capabilities API')
+        }
+        
+        const collections = (!collectionIds || collectionIds.length === 0) ? 
+            await this.storage.list() : 
+            await Promise.all(collectionIds.map(async(collectionName: string) => ({ id: collectionName, fields: await this.schemaInformation.schemaFieldsFor(collectionName) })))
+                
+        const capabilities = this.formatCollectionCapabilities(this.storage.capabilities())
+        const collectionAfterFormat: Collection[] = collections.map((collection) => ({
+            id: collection.id,
+            fields: this.formatFields(collection.fields),
+            capabilities
+        }))
+
+        return { collection: collectionAfterFormat }
+    }
+
+    formatFields(fields: ResponseField[]): Field[] {
```

Comment:
I think that all the methods that shouldn't be used from outside should be private.
(I know we didn't use that before, but maybe it's a good time to start)